### PR TITLE
Office Hour Fix

### PR DESCRIPTION
**Pull request by JasonIsAzn**
### Overview
The bug with office hour was due to removing an old implementation to refetch a user's data. 
Office hour status for would only be fetched after auth was completely loaded (or else it would fail). 

I removed this old implementation due to race condition and some users would be logged out randomly. 

The solution I implemented to this is just to allow for multiple attempts to fetch the user's data.

### Issues
- #480

### CODE REVIEW

File: src/screens/home/Home.tsx [hunk 1]
```diff
@@ -55,6 +55,22 @@ const Home = ({ navigation, route }: NativeStackScreenProps<HomeStackParams>) =>
     const [interestOptionsModal, setInterestOptionsModal] = useState<boolean>(false);
     const [savedInterestLoading, setSavedInterestLoading] = useState<boolean>(false);
 
+    const fetchUserData = async () => {
```
Comment (LucientZ) [inline src/screens/home/Home.tsx:58]: This function exists in 5 different files and is exactly the same. We may possibly want to consider moving it to one file at some point (Maybe somewhere in `firebaseUtils.ts` or `./src/helper`)

File: src/screens/home/Home.tsx [hunk 2]
```diff
@@ -69,6 +85,44 @@ const Home = ({ navigation, route }: NativeStackScreenProps<HomeStackParams>) =>
...
+    const attemptFetchUserDoc = async (user: any, attempts = 0) => {
+        const MAX_ATTEMPTS = 15
+        try {
+            const userDocRef = doc(db, "users", user.uid);
+            const userDoc = await getDoc(userDocRef);
+
+            if (!userDoc.exists()) {
+                if (attempts < MAX_ATTEMPTS) {
+                    console.warn(`Attempt ${attempts + 1} failed. Retrying...`);
+                    await attemptFetchUserDoc(user, attempts + 1);
```
Comment (LucientZ) [inline src/screens/home/Home.tsx:97]: Why is this recursive?

File: src/screens/home/Home.tsx (hunk 1 above)
Comment (JasonIsAzn) [reply src/screens/home/Home.tsx]: [beafe77](https://github.com/TAMUSHPE/MobileApp/pull/490/commits/beafe7719457d540a140c7b62143a738b8abbd66)

Comment (JasonIsAzn) [reply src/screens/home/Home.tsx]: When implementing this, the useContext method isn't allowed in .ts file, so  It seem like a few line of code so i didn't bother too much. 
But extracting it out anyways does make it look better.

File: src/screens/home/Home.tsx (hunk 2 above)
Comment (JasonIsAzn) [reply src/screens/home/Home.tsx]: Ohh the logic for this was really bad.
but fixed now
[4411e2f](https://github.com/TAMUSHPE/MobileApp/pull/490/commits/4411e2f84c8dea4ee3a0ecb14aa1e5e28d2df34f)

refactor with extracted fetchAndStoreUserFunction
fb17f01e8c2445b012eedb50679bd81547704b3d

File: src/screens/home/Home.tsx (hunk 1 above)
Comment (LucientZ) [reply src/screens/home/Home.tsx]: > When implementing this, the useContext method isn't allowed in .ts file, so It seem like a few line of code so i didn't bother too much. But extracting it out anyways does make it look better.

Oh, I see. Fair enough.

**Review by LucientZ — APPROVED** [review]
Looks good now!

**Pull request merged**